include/deal.II/matrix_free/fe_point_evaluation.h

```diff
@@ -830,6 +839,8 @@ FEPointEvaluation<n_components, dim, spacedim, Number>::setup(
   AssertIndexRange(first_selected_component + n_components,
                    fe->n_components() + 1);
 
+  shapes.reserve(2000);
```

Comment:
Why 2000?

Reply:
I wanted to choose it large enough, that `reinit()` does not need to allocate any more memory. We had 200 internally before for one vectorized unit point batch, now we precompute it for all batches so I multiplied it by 10. This value is up to discussion! What do you think?

Reply:
You should not need such a big array. Typical polynomial degrees are between 1 and 6, most often 1-3, so one of the dimension is already between 2 and 7. Obviously, we might have a few hundreds of points, but with vectorization enabled I think even the second loop is often less than 50-100 in length. So I suggest to use a lower value of 100.

Reply:
Then, let's go with 100 for now, and if we see additional allocations we can increase it if necessary.

Reply:
For most cases, we would not re-allocate very often, because we have this code:
https://github.com/dealii/dealii/blob/6a5df1fc493c96c35d52715a85e194175be0055b/include/deal.II/base/aligned_vector.h#L1366-L1370
So we would re-allocate either once if the number of points is constant, but at most `log2(N)` times (in practice: 3-4 times), which should be negligible compared to the number of allocations we do in the constructor.